libraries/Microsoft.Bot.Builder.BotFramework/BotFrameworkBotServer.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Servers;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.BotFramework
+{
+    public class BotFrameworkBotServer : BotServer
+    {
+        private readonly SimpleCredentialProvider _credentialProvider;
+        private readonly MicrosoftAppCredentials _credentials;
+        private readonly HttpClient _httpClient; 
+
+        public BotFrameworkBotServer(IConfiguration configuration, HttpClient httpClient=null) : base()
+        {
+            _httpClient = httpClient ?? new HttpClient();
+            _credentialProvider = new ConfigurationCredentialProvider(configuration);
+            _credentials = new MicrosoftAppCredentials(_credentialProvider.AppId, _credentialProvider.Password);                                   
+        }
+
+        public BotFrameworkBotServer(string appId, string appPassword, HttpClient httpClient = null) : base()
+        {
+            _httpClient = httpClient ?? new HttpClient();
+            _credentials = new MicrosoftAppCredentials(appId, appPassword);
+            _credentialProvider = new SimpleCredentialProvider(appId, appPassword);
+        }
+
+        public new BotFrameworkBotServer Use(Middleware.IMiddleware middleware)
+        {
+            base._middlewareSet.Use(middleware);
+            return this;
+        }
+
+        public async Task ProcessActivty(string authHeader, Activity activity, Func<IBotContext,Task> callback)
```

Comment:
If we do introduce a BotController class we could more the auth into that. That way the responsibility of this layer is to tie the inbound and outbound worlds together and so host the pipeline.
(This is not something we need to change for this checkin.)